test/Unit/Task/SecurityCheckerEnlightnTest.php

```diff
@@ -107,5 +107,16 @@ public function provideExternalTaskRuns(): iterable
                 './composer.lock',
             ]
         ];
+
+        yield 'with_allow_list' => [
+            ['allow_list' => ['allow_advisory_1', 'allow_advisory_2']],
+            $this->mockContext(RunContext::class, ['composer.lock']),
+            'security-checker',
+            [
+                'security:check',
+                './composer.lock',
+                '--allow-list=allow_advisory_1,allow_advisory_2'
```

Comment:
This is not how addArgumentArrayWithSeparatedValue works.
If you want this output, you could use something like:

```php
addOptionalCommaSeparatedArgument('--allow-list=%s', $config['allow_list']);
```